src/main/java/org/openjdk/jextract/impl/NameMangler.java

```diff
@@ -327,7 +332,16 @@ private static boolean isJavaTypeName(String name) {
                 "MemoryLayout",
                 "Arena", "NativeArena", "MemorySegment", "ValueLayout"
                     -> true;
-            default -> false;
+            default -> {
+                try {
+                    // java.lang package is auto-imported. Check if we've clash
+                    // of the given identifier with any of the java.lang.* types.
+                    Class.forName("java.lang." + name);
```

Comment:
1. Should we cache the results?
2. This detects package-private classes in java.lang too, which might not be desired. Maybe `yield clazz.accessFlags().contains(AccessFlag.PUBLIC)`

Reply:
Will fix